doc/paikannuspohjan_kontekstit.md

```diff
@@ -0,0 +1,321 @@
+niitä voidaan edelleen muokata ja täydentää ennen kun ne julkaistaan normaalin luonnoksen tapaan viralliseksi. Tässä
+kohtaa myös suunnitelmajulkaisu päivitetään valmistuneeksi.
+
+## ID käsittely ja viittaukset kontekstien välillä
+
+Eri kontekstien versiot samasta käsitteestä ovat kannassa omia rivejään ja siten niillä on omat ID:nsä. Käyttöliittymän
+kannalta käsittellä on kuitenkin aina vain yksi ID ja lisätietona konteksti jossa sitä tarkastellaan. Tuo yksi ID on
+käsitteenä nimeltään virallinen ID (official ID). Käytännössä se on käsitteen ensimmäisen ilmenemän ID, riippumatta
+missä kontekstissa se ensimmäisen kerran syntyi. Koska tuo ID ei muutu käsitteen siirtyessä eri kontekstien välillä,
+käyttöliittymä voi helposti siirtyä tarkastelemaan koko paikannuspohjaa eri konteksteissa.
+
+Myös käsitteiden keskinäiset viittaukset tehdään aina tällä virallisella ID:llä, minkä johdosta noita viittauksia ei
+tarvitse korjata silloin kun käsiteestä tehdään uusi versio toiseen kontekstiin -- virallinen ID säilyy aina samana.
+Koska luonnos ja suunnitelmat rakentuvat virallisen paikannuspohjan päälle, tämä tarkoittaa että niiden sisältämät
+käsitteet voivat viitata virallisilla ID:llä sekä oman kontekstin käsitteisiin että virallisiin käsitteisiin.
+
+Lopullinen rivi jota viite tarkoittaa pitää siis aina tulkita huomioiden myös konteksti. Jos esimerkiksi viralliseen
+käsitteeseen tulee muutos luonnoskontekstiin, viite olioon säilyy muuttumattomana ja osoittaa nyt luonnosversioon.
+Vastaavasti jos käsite poistuu luonnoskontekstista, viite tarkoittaa nyt virallisesta paikannuspohjasta tulevaa oliota.
+
+Käsitteiden elinkaaria uuden luonnissa tai olemassaolevan muutoksissa kuvataan alla tarkemmin eri skenaarioille
+kontekstit huomioiden. Huomattavaa on, että ensimmäisenä syntynyt olio (tietokantarivi) säilyy ja siirtyy lopulta
+viralliseen paikannuspohjaan, mutta muokattaessa muutokset tehdään kopioon, josta data kopioidaan alkuperäiselle
+julkaisussa. Tämä varmistaa että virallinen ID säilyy muita viittaajia varten kaikissa ketjuissa.
+
+### Kaavioiden merkinnät
+
+```mermaid
+graph TD
+    subgraph legendColor [Värikoodit]
+        legendDraft(Luonnos olio)
+        legendOfficial(Virallinen olio)
+        legendPlan(Suunniteltu olio)
+        legendDraft ~~~ legendOfficial ~~~ legendPlan
+    end
+    subgraph legendSymbol [Symbolit]
+        legendOfficialId("Virallinen ID\n❖ id=x")
+    end
+    subgraph legendLine [Viivat]
+        A -- Viite olioiden välillä --> B
+        Cv1 == Olio siirtyy ==> Cv2
+        D -. Olion data kopioidaan .-> E
+    end
+    classDef draftObject fill: lightyellow, stroke: black, color: black;
+    classDef officialObject fill: lightgreen, stroke: black, color: black;
+    classDef planObject fill: pink, stroke: black, color: black;
+    class legendDraft draftObject
+    class legendOfficial officialObject
+    class legendPlan planObject
+    linkStyle 3 stroke: grey, stroke-dasharray: 5 5
```

Comment:
Muut lisätyylit voi tehdä kivasti noilla luokilla ja liittää ne vain olioihin, mutta linkkien tyylit pitää osoittaa oikeaan viivaan indeksillä. Luonnollisesti ne levähtää jos väliin pirtää yhden viivan lisää ja oikean indeksin löytäminen onnistuu oikeastaan vain trial-and-error mekanismilla.